src/main/java/com/example/javawebcourse/config/FeatureToggleProperties.java

```diff
@@ -0,0 +1,22 @@
+package com.example.javawebcourse.config;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.Map;
+
+@Data
+@NoArgsConstructor
+@Configuration
+@ConfigurationProperties(prefix = "application.feature")
+public class FeatureToggleProperties {
+
+    Map<String, Boolean> toggles;
+
+    public boolean check(String featureToggle){
+        return toggles.getOrDefault(featureToggle, false);
+    }
+
+}
```

Comment:
```suggestion
@AllArgsConstructor(onConstructor_ = @ConstructorBinding)
@ConfigurationProperties(prefix = "application.feature")
public class FeatureToggleProperties {

    private Map<String, Boolean> toggles;

    public boolean isEnabled(String featureToggle){
        return toggles.getOrDefault(featureToggle, false);
    }

}
```

It is generally good idea to keep configuration properties immutable. Also avoid exposing direct getters on collections since it breaks encapsulation by allowing to modify underlying collection. Your approach here is the best one, but use @ConstructorBinding to initalize properties with constructor and then keep them frozen